ui/src/App.js

```diff
@@ -3,25 +3,30 @@ import { Button, Box, Typography, AppBar, Toolbar } from '@material-ui/core';
 import LandingPage from './LandingPage'
 import SurveyDetails from './SurveyDetails'
 import CollectionExerciseDetails from './CollectionExerciseDetails'
-import CaseSearch from './CaseSearch';
 import CaseDetails from './CaseDetails';
+import SurveyCaseSearch from './SurveyCaseSearch'
 
 class App extends Component {
   state = {
     selectedSurveyId: null,
     selectedSurveyName: '',
+    collectionExercises: [],
     selectedCollexId: null,
     selectedCollexName: '',
-    caseSearchActive: false,
     caseSearchResults: [],
+    caseSearchTerm: '',
+    caseSearchDesc: '',
```

Comment:
it feels weird to have these in here, but not the other search options, if they're here just to preserve them when the user switches between screens

Reply:
I think I know what this is for... it's for the `Results for sample data containing "xxx":` bit, right? If so, ignore the comment... I understand where you're coming from and why it's there.

But.

You can guarantee that some smart alec is going to say "why does it forget the search settings when you click on case details"... but I don't see that as any reason not to approve this PR

Reply:
I'll have a quick look at remembering those details on the page too, no reason it can't now. 
Also, do you think it would be nicer to bundle the cases list, term and description into a single object in the top level state? Feels messy to me to keep adding stuff at the top level to facilitate small feature changes lower.

Reply:
the Rolls Royce solution would probably bundle the entire state of the search boxes into a single JSON blob, like you say. Bit of a pain, perhaps, but it's definitely something that could come to bite us with users asking why their search is forgotten when they click on case details